Android/toonDra/app/src/main/java/edu/kaist/mskers/toondra/PreviewModeFragment.java

```diff
@@ -0,0 +1,175 @@
+    leftLinear.removeAllViews();
+    final ArrayList<Thumbnail> thumbnailArrayList = new ArrayList<Thumbnail>();
+
+    /* Download webtoon list by day and set first preview link */
+    Thread thread = new Thread(new Runnable() {
+      @Override
+      public void run() {
+        NaverToonInfo[] webtoons = ((MainActivity)getActivity()).getCurrentWebtoonsByDay(day);
+        for (int i = 0; i < webtoons.length; i++) {
+          final Thumbnail thumb = new Thumbnail(getActivity());
+          thumb.initView(webtoons[i]);
+          thumbnailArrayList.add(thumb);
+
+          final String firstEpisodeUrl = thumb.getFirstEpisodeUrl();
+
+          View.OnClickListener onClickListener = new View.OnClickListener() {
+            public void onClick(View view) {
+              if (SystemClock.elapsedRealtime() - lastClickTime < 1000) {
+                return;
+              }
+              lastClickTime = SystemClock.elapsedRealtime();
+
+              if (currentWebtoon != null) {
+                currentWebtoon.thumbName.setBackgroundColor(ContextCompat
+                    .getColor(getActivity(), R.color.waitButtonColor));
+              }
+              thumb.thumbName.setBackgroundColor(ContextCompat
+                  .getColor(getActivity(), R.color.activeButtonColor));
+              currentWebtoon = thumb;
+              rightLinear.removeAllViews();
+              startFirstEpisodePreview(firstEpisodeUrl);
+            }
+          };
+
+          View.OnLongClickListener onLongClickListener = new View.OnLongClickListener() {
+            public boolean onLongClick(View view) {
+              Log.e("longClick", "ok");
+              Intent intent = new Intent(getActivity(), EpisodeListPage.class);
+              startActivity(intent);
+              return false;
+            }
+          };
+
+          thumb.thumbLayout.setOnClickListener(onClickListener);
+          thumb.thumbLayout.setOnLongClickListener(onLongClickListener);
+
+          getActivity().runOnUiThread(new Runnable() {
+            @Override
+            public void run() {
+              leftLinear.addView(thumb);
+            }
+          });
+
+        }
+      }
+    });
+    thread.start();
+  }
+
+  /**
+   * Add the first episode to the right side of the main view by giving first episode url.
+   */
+  private void startFirstEpisodePreview(final String firstEpisodeUrl) {
+    final LinearLayout rightLinear = (LinearLayout) getView().findViewById(R.id.rightLinearLayout);
+    final Thread pageThread = new Thread(new Runnable() {
+      @Override
+      public void run() {
+        try {
+          Element wtPage;
+          wtPage = Jsoup.connect(firstEpisodeUrl)
+              .userAgent("Mozilla/5.0 (Windows NT 6.1; WOW64; rv:23.0)"
+                  + "Gecko/20100101 Firefox/23.0")
+              .get();
+          Element wtViewer = wtPage.getElementsByClass("wt_viewer").first();
+
+          for (Element imgLink : wtViewer.children()) {
```

Comment:
I think NullPointerException only occurs when the user accesses the 19 years old webtoon. (It is blocked due to permission problem.) I will put alert message for this.